lib/app/features/chat/providers/user_chat_relays_provider.c.dart

```diff
@@ -1,39 +1,73 @@
 // SPDX-License-Identifier: ice License 1.0
 
 import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/exceptions/exceptions.dart';
+import 'package:ion/app/features/auth/providers/auth_provider.c.dart';
 import 'package:ion/app/features/nostr/model/action_source.dart';
 import 'package:ion/app/features/nostr/providers/nostr_cache.c.dart';
 import 'package:ion/app/features/nostr/providers/nostr_notifier.c.dart';
 import 'package:ion/app/features/user/model/user_chat_relays.c.dart';
+import 'package:ion/app/features/user/model/user_relays.c.dart';
+import 'package:ion/app/features/user/providers/user_relays_manager.c.dart';
 import 'package:nostr_dart/nostr_dart.dart';
 import 'package:riverpod_annotation/riverpod_annotation.dart';
 
 part 'user_chat_relays_provider.c.g.dart';
 
-///
-/// This provider is used to fetch the user chat relays for a given pubkey.
-/// It is different from the [UserRelays] provider because it is used to fetch the user chat relays
-///
 @riverpod
-Future<UserChatRelaysEntity?> userChatRelays(
-  Ref ref,
-  String pubkey,
-) async {
-  final entity = ref.watch(
+Future<UserChatRelaysEntity?> userChatRelays(Ref ref, String pubkey) async {
+  final cached = ref.watch(
     nostrCacheProvider.select<UserChatRelaysEntity?>(
       cacheSelector(UserChatRelaysEntity.cacheKeyBuilder(pubkey: pubkey)),
     ),
   );
-  if (entity != null) {
-    return entity;
-  }
+  if (cached != null) return cached;
 
   final requestMessage = RequestMessage()
     ..addFilter(
       RequestFilter(kinds: const [UserChatRelaysEntity.kind], authors: [pubkey]),
     );
 
-  return ref
-      .watch(nostrNotifierProvider.notifier)
-      .requestEntity<UserChatRelaysEntity>(requestMessage, actionSource: ActionSourceUser(pubkey));
+  return ref.watch(nostrNotifierProvider.notifier).requestEntity<UserChatRelaysEntity>(
+        requestMessage,
+        actionSource: ActionSourceUser(pubkey),
+      );
+}
+
+@riverpod
+class UserChatRelaysManager extends _$UserChatRelaysManager {
+  @override
+  FutureOr<void> build() {}
```

Comment:
This Notifier doesn't hold any state, and you don't watch it's `AsyncValue` (like loading or error), please, refactor to not use the Notifier.

Reply:
fixed